[spec] Update typeshed language to conform to reality (#1571)
python/typing: docs/spec/distributing.rst

  Any identifier not defined in the stub is therefore assumed to be of type
  ``Any``.

The Typeshed Project
^^^^^^^^^^^^^^^^^^^^

> The `typeshed project <https://github.com/python/typeshed>`_ contains type
> annotations for the standard library and a selection of third-party libraries
> that don't ship their own type annotations or stubs. Policies regarding the

Reviewer: ```suggestion
The `typeshed project <https://github.com/python/typeshed>`_ contains type
stubs for the standard library (vendored or handled specially by type checkers)
and type stubs for third-party libraries that don't ship their own type information
(typically distributed via PyPI). Policies regarding the
```

> stubs collected there are decided separately and reported in the project's

Reviewer 2: reported -> described? "Reporting a policy" somehow doesn't sound right.

   bundled with the package SHOULD be used (be they in ``.pyi`` type
   stub files or inline in ``.py`` files).

> 5. Typeshed (if used) - Provides the stdlib types.

Reviewer: ```suggestion
5. Typeshed - only for modules in the standard library.
```
